### PR TITLE
snapmgr, backends: maybe restart & security backend options

### PR DESCRIPTION
**Pull request by stolowski**
Small followup to #8046 to address final remarks from @pedronis :
- Make maybeRestart a method of snapmgr (and avoid restart in preseed mode there rather than when calling it).
- Introduce SecurityBackedOptions to pass preseed flag to Initialize methods of backend (adjust apparmor setup accordingly).

### CODE REVIEW

File: interfaces/dbus/backend.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ import (
 type Backend struct{}
 
 // Initialize does nothing.
-func (b *Backend) Initialize() error {
+func (b *Backend) Initialize(opts *interfaces.SecurityBackendOptions) error {
```
Comment (pedronis) [inline interfaces/dbus/backend.go]: (nitpick) can leave out opts here:  `Initialize(*interfaces.SecurityBackendOptions)`

File: interfaces/ifacetest/backend.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ type TestSetupCall struct {
 }
 
 // Initialize does nothing.
-func (b *TestSecurityBackend) Initialize() error {
+func (b *TestSecurityBackend) Initialize(opts *interfaces.SecurityBackendOptions) error {
```
Comment (pedronis) [inline interfaces/ifacetest/backend.go]: same everywhere

File: overlord/snapstate/handlers_link_test.go [hunk 3]
```diff
@@ -1271,6 +1271,10 @@ func (s *linkSnapSuite) setMockKernelRemodelCtx(c *C, oldKernel, newKernel strin
 }
 
 func (s *linkSnapSuite) TestMaybeUndoRemodelBootChangesUnrelatedAppDoesNothing(c *C) {
+	runner := state.NewTaskRunner(s.state)
+	mgr, err := snapstate.Manager(s.state, runner)
```
Comment (pedronis) [inline overlord/snapstate/handlers_link_test.go]: we have a manager in s.snapmgr, any reason not to use that?

Comment (stolowski) [reply overlord/snapstate/handlers_link_test.go]: Right, I missed that, thanks!

**Pull request merged**